doltpy/cli/dolt.py

```diff
@@ -133,30 +127,37 @@ def __init__(self, name: str, url: str):
         self.url = url
 
 
-class ServerConfig:
+class DoltHubContext:
```

Comment:
handling this in a nice way is relevant for metaflow also. couple thoughts:
- is a generic context manager useful? could have it subclass Dolt, use path-patterns or kwargs to use local or remote repos
- if Dolt can be self-hosted, `DoltRemote` or other namespacing would be inclusive of dolthub + self-hosted
- should a context manager use a temporary dir for cloning?

Reply:
These are really good questions, I have not decided entirely. I do think the context manger is helpful, but I won't pretend I have hit upon the final form of what we should offer.

Reply:
Adding this to a list of Doltpy product/API discussions to have